### PR TITLE
Implement a new scoring system

### LINKED ISSUE #33: Suggestion for project quality metrics
I think this website is a really well-needed resource and thanks for creating it.

One of the key bits for me is the project health feature. I think the simplistic score you have currently is a great start, but I was reminded of the great project quality feature that the Cocoapods (iOS dependency manager) directory has. You can see an example of a libraries quality page here:

https://cocoapods.org/pods/AnimationSugar/quality

By default, it starts a library with a score of 50 and then adds and subtracts from that to get the final score. Things like "being very popular" get an addition and others like "using a GPL licence" (which is arguably incompatible with the Apple App Store) result in a subtraction from the base score.

On the page I linked above, press on the "show metrics not applied" to see the full list of possible metrics. They have also documented how this system works and the theory behind it here:

https://guides.cocoapods.org/making/quality-indexes

Overall, I think a system like this would be a great way to measure the health of React Native libraries as well. Let me know your thoughts and I'm happy to help out if this is a direction you want to move in.

### PR DESCRIPTION
**Pull request by matt-oakes**
I have implemented a new scoring system based on the ideas from the [Cocoapods Quality Metrics](https://guides.cocoapods.org/making/quality-indexes). The new system takes into account:

* Popularity using the subscribers, forks, stars, and npm downloads with items at the start of the list weighted for greater importance as they indicate a more active contribution to the project.
* If the library is marked as recommended (called `goldstar` internally).
* If there are lots of open issues (currently set to 75).
* If there is no licence or a licence which is deemed incompatible with the Apple App Store.
* If the library has been recently updated (last 30 days) or not been updated in a while (over 6 months).

To help compare the previous and new scores, I have produced some statistics using [this script](https://gist.github.com/matt-oakes/55e8cb02d40555f083cb1a468923a96a):

```
Old mean:  73.82
New mean:  55.15
Mean difference:  -18.67

Number of libraries:  258
Match modifier counts:  {
  "Popular": 51,
  "Recently updated": 252,
  "No license": 60,
  "Lots of open issues": 14,
  "Very popular": 10,
  "Recommended": 10,
  "GPL license": 1
}
```

The average score is now lower, but I think this is correct. 50 should be the mark of an average quality library with lower and higher scores being indicative of the quality.

I have also put the data into a Google Sheet to allow you to check over the data more thoroughly:

https://docs.google.com/spreadsheets/d/1GIaIS_JIiz1ile5RknQPF6rjWFbfX8yRaYWB-vH-NQA/edit#gid=904868035

Looking through the scores, most seem to have been reduced because Github cannot determine which licence the library is under. I think this is a valid reason to reduce the score as this would be a barrier to using the library for most people.

All of these values and modifiers and be easily tweaked to fine tune the score.

Let me know what you think!

*Closes #33.*

### CODE REVIEW

**Comment by jimmylee** [comment]
I'm so happy about this, thank you! I'm about to head to the airport and be out of pocket for about 11 hours. I am excited to review the work and land it into our codebase.

**Comment by matt-oakes** [comment]
I have noticed that the issue from #20 is causing the *Recently updated* rule to be applied to libraries which shouldn't qualify. I have asked over in that issue if this is a task which @kelset is currently working on.

**Comment by jimmylee** [comment]
TLDR: LGTM

One concern:

- I'm going to need to update the interface because users have mentioned in the past that having a score beneath 75% is a little upsetting. If I make it clear that 50% represents usability, then I think no one will feel bad.
- I need to make an information page about the scoring at one point.

Again thank you so much for doing this. Sorry it took so long for me to land it :) 🛩

**Comment by matt-oakes** [comment]
@jimmylee No problem. That sounds right to me. It might just be changing it so it looks more like this:

```
|---------------------|---------------------|
Poor-----------------Good---------------Great
```
